src/openms_gui/source/VISUAL/SpectraIdentificationViewWidget.cpp

```diff
@@ -471,11 +407,12 @@ namespace OpenMS
 
     // generate flat list
     int selected_row(-1);
+    // index i is needed, so iterate the old way...
     for (Size i = 0; i < layer_->getPeakData()->size(); ++i)
     {
       QTableWidgetItem* item = nullptr;
 
-      const MSSpectrum & spectrum = (*layer_->getPeakData())[i];
+      const MSSpectrum& spectrum = (*layer_->getPeakData())[i];
```

Comment:
is this safe? you head the lambda last time

Reply:
yes, its safe, because the MS data is not modified by the table